src/transformers/models/esm/modeling_tf_esm.py

```diff
@@ -1216,23 +1212,21 @@ def __init__(self, config, name=None):
         )
 
         self.layer_norm = LayerNormalization(epsilon=config.layer_norm_eps, name="layer_norm")
-
-        self.decoder = None
+        if config.tie_word_embeddings:
+            self.decoder = None
```

Comment:
Wait, when does weight tying happing in the TF models? I just realised that I took it for granted with the `config.tie_weight_embeddings` setting for PyTorch models, because the weights are tied in the `init_weights` call.

Reply:
I don't know if there's a consistent solution across most of our models for this, unfortunately. In ESM weight tying happens at model init - we replace `lm_head.decoder` with the embedding weight matrix (not a copy of it, the actual original weight matrix).